### DISCUSSION TITLE
static paths on nextconfig dependent on environment vars

### DISCUSSION THREAD

**Discussion by albjeremias** [vercel/next.js#52143] [Help] [answered]
### Summary

i am using an environment variable on next.config.js 

https://github.com/helpbuttons/helpbuttons/blob/main/web/next.config.js#L26

this makes so that the routes are configurable before running the app. but I figured out in the build directory that this variables are discarded and built into as a string. is this a bug? or a feature? any workaround?

**Comment by albjeremias** [comment]
a workaround would be to have an yml file or a json file which could be loaded at runtime, how can i do this? thanks

**Comment by icyJoseph** [comment]
> but I figured out in the build directory that this variables are discarded and built into as a string. 

Can you show an example of what you mean by this?

**Reply by albjeremias** [reply]
https://github.com/helpbuttons/helpbuttons/blob/main/web/next.config.js#L26

I mean this uses the variable process.env.API_URL

but the value of the rewrite is only changed when i run `yarn build` and i would like to be when i run `yarn start` because i want to use this app in a docker image.. where the environment variables are configured...

```
 rewrites: async () => {
        return [
          {
            source: '/geoapify/:path*',
            destination: 'https://api.geoapify.com/:path*',
          },
          {
            source: '/api/:path*',
            destination: `${process.env.API_URL ? process.env.API_URL : 'http://api:3001/' }:path*`
          },
          {
            source: '/opencage/:path*',
            destination: 'https://api.opencagedata.com/geocode/v1/:path*',
          },
        ]
    },
```

**Reply by icyJoseph** [reply]
Aha, so those are getting processed into the routes manifest file, right? 

I'd say then, that you are probably better off using middleware, which supports both rewrites and redirects.

**Reply by albjeremias** [reply]
the thing is.. this issue bugged me for many days.. cause u known in docker we use environment variables to make a lot of stuff... and is very weird this behavior, but it seems is not only nextjs but the whole webpack ecosystem. 

for me it goes against my usage of environment variables, i normally never used them for building, but only for runtime.
anyway can u point me out some middleware tho exemplify that? thanks

**Reply by icyJoseph** [reply, marked as answer]
Yea, normally it would work, but the routes manifest was the exception :( 

- https://nextjs.org/docs/app/building-your-application/routing/middleware#conditional-statements

Same docs under pages router: https://nextjs.org/docs/pages/building-your-application/routing/middleware#conditional-statements

**Reply by albjeremias** [reply]
great! :) but im using the pages router! 

so its this one:
https://nextjs.org/docs/pages/building-your-application/routing/middleware

**Reply by icyJoseph** [reply]
Yeah it's the same for both. Still not used to the new docs 😅
